src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java

```diff
@@ -634,11 +635,39 @@ public void registerEvents(@NotNull Listener listener, @NotNull Plugin plugin)
 		{
 			throw new IllegalPluginAccessException("Plugin attempted to register " + listener + " while not enabled");
 		}
-
+		addListener(listener, plugin);
 		for (Map.Entry<Class<? extends Event>, Set<RegisteredListener>> entry : plugin.getPluginLoader().createRegisteredListeners(listener, plugin).entrySet())
 		{
 			getEventListeners(getRegistrationClass(entry.getKey())).registerAll(entry.getValue());
 		}
+
+	}
+
+	private void addListener(Listener listener, Plugin plugin)
+	{
+		List<Listener> l  = listeners.getOrDefault(plugin.getName(), new ArrayList<>());
+		if (l.contains(listener))
+		{
+			return;
+		}
+		l.add(listener);
+		listeners.put(plugin.getName(), l);
```

Comment:
```suggestion
		if (!l.contains(listener))
		{
			l.add(listener);
			listeners.put(plugin.getName(), l);
		}
```
This seems a bit shorter and more readable imo